test/jasmine/tests/cartesian_interact_test.js

```diff
@@ -2244,3 +2247,100 @@ describe('Event data:', function() {
         .then(done);
     });
 });
+
+describe('Cartesian plots with css transforms', function() {
+    var gd;
+    var eventRecordings = {};
+
+    beforeEach(function() {
+        eventRecordings = {};
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    function _getLocalPos(element, point) {
+        var bb = element.getBoundingClientRect();
+        return [
+            bb.left + point[0],
+            bb.top + point[1]
+        ];
+    }
+
+    function _hover(pos) {
+        return new Promise(function(resolve, reject) {
+            var localPos = _getLocalPos(gd, pos);
+            gd.once('plotly_hover', function(d) {
+                Lib.clearThrottle();
+                resolve(d);
+            });
+
+            mouseEvent('mousemove', localPos[0], localPos[1]);
+
+            setTimeout(function() {
+                reject('plotly_hover did not get called!');
+            }, 100);
+        });
+    }
+
+    function _unhover(pos) {
+        var localPos = _getLocalPos(gd, pos);
+        mouseEvent('mouseout', localPos[0], localPos[1]);
+    }
+
+    function _hoverAndAssertEventOccurred(point, label) {
+        return _hover(point)
+        .then(function() {
+            expect(eventRecordings[label]).toBe(1);
+        })
+        .then(function() {
+            _unhover(point);
+        });
+    }
+
+    function transformPlot(gd, scale, transX, transY) {
```

Comment:
Let's simply pass `transformString` to this function.
```
function transformPlot(gd, transformString) {
```
That way we could easily test various combinations of translates and scales.

Reply:
You're right, that's better